components/CheckInOutForm/index.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  Autocomplete,
+  Box,
+  FormControl,
+  IconButton,
+  TextField,
+  Typography,
+} from '@mui/material'
+import { DateTimePicker } from '@mui/x-date-pickers'
+import dayjs, { Dayjs } from 'dayjs'
+import React from 'react'
+import { ItemDefinition, User } from '../../utils/types'
+import AddOutlinedIcon from '@mui/icons-material/AddOutlined'
+import RemoveOutlinedIcon from '@mui/icons-material/RemoveOutlined'
+import colors from '../../utils/colors'
+
+interface Props {
+  kioskMode: boolean
+  users: User[]
+  itemDefinitions: ItemDefinition[]
+}
+
+function CheckInOutForm({ kioskMode, users, itemDefinitions }: Props) {
+  const [date, setDate] = React.useState<Dayjs | null>(dayjs(new Date()))
+  const [quantity, setQuantity] = React.useState<number>(1)
+  const [selectedStaff, setSelectedStaff] = React.useState<User | null>()
+  const [selectedItemDefinition, setSelectedItemDefinition] =
+    React.useState<ItemDefinition | null>()
+
+  return (
+    <>
+      <FormControl fullWidth>
+        {/* Staff member, Date, and Item input fields */}
+        {kioskMode && (
+          <Autocomplete
+            options={users.map((user) => user.name)}
+            renderInput={(params) => (
+              <TextField {...params} label="Staff Member" />
+            )}
+            onChange={(_e, name) =>
+              setSelectedStaff(users.find((user) => name === user.name))
+            }
+          />
+        )}
+        <Box sx={{ marginTop: 4 }}>
+          <DateTimePicker
+            label="Date"
+            value={date}
+            onChange={(date) => setDate(date)}
+            renderInput={(params) => <TextField {...params} fullWidth />}
+          />
+        </Box>
+        <Autocomplete
+          options={itemDefinitions.map((itemDefinition) => itemDefinition.name)}
+          sx={{ marginTop: 4 }}
+          renderInput={(params) => <TextField {...params} label="Item" />}
+          onChange={(_e, name) =>
+            setSelectedItemDefinition(
+              itemDefinitions.find(
+                (itemDefinition) => name === itemDefinition.name
+              )
+            )
+          }
+        />
+        {/* Quantity input field and +- buttons */}
+        <Box sx={{ display: 'flex', alignSelf: 'center', marginTop: 3 }}>
```

Comment:
suggestion (feel free to disagree): I think this should be its own component for two reasons.
1. If there is a requirement change and we do need to use this somewhere else, it makes it easier to do that
2. It would make this component a lot cleaner and easier to parse. Half the code in this component is for building that quantity field which clutters up the component quite a bit.

Again, just my opinion let me know how you feel

Reply:
Agree. Made a new component